### PR TITLE
158627390 Refactor create asset category endpoint and document api

### PR DESCRIPTION
**Pull request by SEUNAGBEYE**
with this data
 `{
	"name": "Headset",
	"attributes": [
              {"label": "brand", "is_required": "true", "input_control": "text", "choices": "null"}, 
              {"label": "color", "is_required": "true", "input_control": "dropdown", "choices": "blue,red,black"}
         ]
}` and an asset category should be created
>- To get errors remove the `Authorization` header or tamper with the request data

### CODE REVIEW

File: api/utilities/helpers.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+def to_camel_case(data):
+    """
+    Converts space seperated character to camel case
+    """
+    
+    first_letter = data[0].lower()
+    data_without_first_letter = data.title()[1:].replace(' ', '')
+    data = first_letter + data_without_first_letter
+
+    return data
```
Comment (Dheavyman) [inline api/utilities/helpers.py]: Add a new line at the end of the file

File: api/utilities/validators/input_control_validator.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+from api.utilities.enums import InputControlChoiceEnum
+from ..messages.error_messages import serialization_errors
+
+def input_control_validation(value):
+    input_controls = InputControlChoiceEnum.get_all_choices()
+    if value.lower() not in input_controls:
+        raise ValidationError({'message': serialization_errors['input_control']})
```
Comment (Dheavyman) [inline api/utilities/validators/input_control_validator.py]: Add newline to the end of file

**Comment by SEUNAGBEYE** [comment]
@brayoh All feedback implemented, could this be merged please so I can rebase the develop and make necessary changes to the `update` branch and push.

File: api/views/asset_category.py [hunk 3]
```diff
@@ -43,6 +42,8 @@ def post(self):
 
             asset_category.attributes = attributes
             attributes = attributes_schema.dump(attributes).data
+        else:
+            raise ValidationError({'message': 'Please provide at least one attribute'})
```
Comment (purpose50) [inline api/views/asset_category.py]: Is it possible to have this error message in the error_message file in the utilities folder.

Comment (SEUNAGBEYE) [reply api/views/asset_category.py]: @purpose50 Done.

**Comment by SEUNAGBEYE** [comment]
@brayoh PR ready to be merged

File: api/utilities/helpers.py (hunk 1 above)
Comment (brayoh) [reply api/utilities/helpers.py]: @SEUNAGBEYE why not go with husky for the above functionality, it's more solid considering it's widely used and also it handles all edge cases

Comment (SEUNAGBEYE) [reply api/utilities/helpers.py]: @brayoh Could you please help with the link to husky documentation?

Comment (SEUNAGBEYE) [reply api/utilities/helpers.py]: @brayoh Done.

**Comment by brayoh** [comment]
nice!

**Pull request merged**